spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Unsqueeze.scala

```diff
@@ -32,46 +32,51 @@ import scala.reflect.ClassTag
 
 @SerialVersionUID(- 5180889605872472241L)
 class Unsqueeze[T: ClassTag](
-  val pos: Int,
+  val pos: Array[Int],
   var numInputDims: Int = Int.MinValue
 )(implicit ev: TensorNumeric[T]) extends AbstractModule[Tensor[_], Tensor[_], T]  {
 
   def setNumInputDims(numInputDims: Int): Unit = {
     this.numInputDims = numInputDims
   }
 
-  private def getActualPosition(input: Tensor[_]) : Int = {
-    val dim = if (pos <= 0) {
-      input.dim() + pos + 1
-    } else {
-      pos
-    }
-
-    // get valid dimension offset for batchMode (if any)
-    val inputDim = input.dim() // data batch dim
-    numInputDims = if (numInputDims != Int.MinValue) numInputDims else inputDim // feature map dim
-    val offsetDim = inputDim - numInputDims
-    require(offsetDim >= 0, "input feature map dim (numInputDims) must be <= input:dim()," +
-      s" input feature map dim ${numInputDims}, inputdim ${inputDim}")
-
-    // the actual position; clearer error message for batchMode (if any)
-    val actualPos = dim + offsetDim
-    require(actualPos >= 1 && actualPos <= (inputDim + 1), s"Invalid position: $pos. " +
-      s"input:dim() is $input, input feature map dim (numInputDims) is $numInputDims.")
-
-    actualPos
+  private def getActualPosition(input: Tensor[_]) : Array[Int] = {
+    pos.foreach(pos => {
+      val dim = if (pos <= 0) {
+        require(pos <= 0, s"invalid input: index starts from 1, " +
+          s"creating a singleton dim at the end of the input tensor")
+        input.dim() + pos + 1
+      } else {
+        pos
+      }
+      // get valid dimension offset for batchMode (if any)
+      val inputDim = input.dim() // data batch dim
+      numInputDims = if (numInputDims != Int.MinValue) numInputDims else inputDim // feature map dim
+      val offsetDim = inputDim - numInputDims
+      require(offsetDim >= 0, "input feature map dim (numInputDims) must be <= input:dim()," +
+        s" input feature map dim ${numInputDims}, inputdim ${inputDim}")
+
+      // the actual position; clearer error message for batchMode (if any)
+      val actualPos = dim + offsetDim
+      require(actualPos >= 1 && actualPos <= (inputDim + 1), s"Invalid position: $pos. " +
+        s"input:dim() is $input, input feature map dim (numInputDims) is $numInputDims.")
+      actualPos
+    })
+    for {
+      i <- 1 to pos.length
+      j <- i to pos.length
+    } if (pos(j) > pos(i)) { pos(j) = pos(j) + 1}
```

Comment:
Please add comment/explanation to this code.
j <- i + 1 to pos.length would make it easier to understand?

Reply:
Yes, ok